### PR TITLE
Fixes for Windows build with new MSYS2

### PR DESCRIPTION
**Pull request by AnHeuermann**
### Related Issues

We might want to use a newer version of msys in OMDev, so @mahge and myslef tried to get OpenModelica compiling on both the newer and the older version from OMDev.

### Purpose

  - Minor changes to CMake and make files
  - Should be possible to build with old an new msys using CMake.

### CODE REVIEW

File: OMEdit/OMEditGUI/OMEditGUI.win.config.pri [hunk 1]
```diff
@@ -48,8 +48,10 @@ CONFIG(release, debug|release) { # release
   # win32 vs. win64
   contains(QT_ARCH, i386) { # 32-bit
     LIBS += -L$$(OMDEV)/tools/msys/mingw32/lib/binutils -L$$(OMDEV)/tools/msys/mingw32/bin
+    QMAKE_CXXFLAGS += -I$$(OMDEV)/tools/msys/mingw32/include/binutils
   } else { # 64-bit
     LIBS += -L$$(OMDEV)/tools/msys/mingw64/lib/binutils -L$$(OMDEV)/tools/msys/mingw64/bin
+    QMAKE_CXXFLAGS += -I$$(OMDEV)/tools/msys/mingw64/include/binutils
```
Comment (mahge) [inline OMEdit/OMEditGUI/OMEditGUI.win.config.pri:54]: We should do this for OMEditLib as well. Probably here 

https://github.com/OpenModelica/OpenModelica/blob/2c2d71103bb0c7c67417ff21a18581fb4b820ae8/OMEdit/OMEditLIB/OMEditLIB.pro#L74

but I am not sure.

Comment (AnHeuermann) [reply OMEdit/OMEditGUI/OMEditGUI.win.config.pri]: Also we need to use `QMAKE_CFLAGS` instead of  `QMAKE_CXXFLAGS`.

Comment (mahge) [reply OMEdit/OMEditGUI/OMEditGUI.win.config.pri]: Is there no `QMAKE_CPPFLAGS`?

Comment (AnHeuermann) [reply OMEdit/OMEditGUI/OMEditGUI.win.config.pri]: At least none is used.

**Comment by AnHeuermann** [comment]
@adeas31 How did you organice your C compiler flags in OMEditLib?
I changed OMEdit/OMEditLIB/CrashReport/backtrace.h to always `#include <bfd.h>` and let the build environemt handle the location of the include directory. It should be working now, but I'm not sure if you like the way I did it.

**Comment by adeas31** [comment]
Doesn't it work if you just add the path to INCLUDEPATH variable.
And perhaps it only needed for OMEditLib.

**Comment by AnHeuermann** [comment]
I added the include directory to `INCLUDEPATH` for OMEditLib, but had to use `QMAKE_CXXFLAGS` in OMEditGui. I hope this is more or less how you want it to be in OMEdit. (And that Jenkins will compile as well...)

**Comment by adeas31** [comment]
Still not sure why `QMAKE_CXXFLAGS`? Its not a flag its an include path. And should be added where other include paths are done in `OMEditGUI.pro` and `OMEditLib.pro` files.

What happens if you don't add it to OMEditGUI?

**Comment by AnHeuermann** [comment]
Without `QMAKE_CXXFLAGS += -I$$(OMDEV)/tools/msys/mingw64/include/binutils` the compiler flag `-I` is missing from the C++ compiler call:

```bash
make[4]: Entering directory '/d/workspace/OM/OpenModelicaMakefiles/OMEdit/OMEditGUI'
clang++ -c -g -DUA_DYNAMIC_LINKING -isystem ../ -isystem ../OMEditLIB -isystem ../OMEditLIB/CrashReport -isystem D:/workspace/OM/OpenModelicaMakefiles/build/include/omc/c -O2 -Wall -Wextra -Wextra -fexceptions -DUNICODE -D_UNICODE -DWIN32 -DMINGW_HAS_SECURE_API=1 -DIMPORT_INTO=1 -DQT_NO_DEBUG -DQT_WEBKITWIDGETS_LIB -DQT_WEBKIT_LIB -DQT_SVG_LIB -DQT_OPENGL_LIB -DQT_PRINTSUPPORT_LIB -DQT_WIDGETS_LIB -DQT_GUI_LIB -DQT_XMLPATTERNS_LIB -DQT_NETWORK_LIB -DQT_XML_LIB -DQT_CONCURRENT_LIB -DQT_CORE_LIB -DQT_NEEDS_QMAIN -I. -I.. -I../OMEditLIB -I../OMEditLIB/CrashReport -I../../build/include/omc/c -IC:/OMDev/tools/msys/mingw64/include/QtWebKitWidgets -IC:/OMDev/tools/msys/mingw64/include/QtWebKit -IC:/OMDev/tools/msys/mingw64/include/QtSvg -IC:/OMDev/tools/msys/mingw64/include/QtOpenGL -IC:/OMDev/tools/msys/mingw64/include/QtPrintSupport -IC:/OMDev/tools/msys/mingw64/include/QtWidgets -IC:/OMDev/tools/msys/mingw64/include/QtGui -IC:/OMDev/tools/msys/mingw64/include/QtXmlPatterns -IC:/OMDev/tools/msys/mingw64/include/QtNetwork -IC:/OMDev/tools/msys/mingw64/include/QtXml -IC:/OMDev/tools/msys/mingw64/include/QtConcurrent -IC:/OMDev/tools/msys/mingw64/include/QtCore -Irelease -I/include -IC:/OMDev/tools/msys/mingw64/share/qt5/mkspecs/win32-g++  -o release/main.o main.cpp
In file included from main.cpp:52:
../OMEditLIB\CrashReport/backtrace.h:34:10: fatal error: 'bfd.h' file not found
#include <bfd.h>
         ^~~~~~~
1 error generated.
make[4]: *** [Makefile.Release:156: release/main.o] Error 1
make[4]: Leaving directory '/d/workspace/OM/OpenModelicaMakefiles/OMEdit/OMEditGUI'
make[3]: *** [Makefile:45: release] Error 2
make[2]: *** [Makefile:76: sub-OMEditGUI-make_first] Error 2
make[1]: *** [Makefile.omdev.mingw:56: OMEdit] Error 2
make: *** [Makefile.omdev.mingw:93: omedit] Error 2
```

**Comment by adeas31** [comment]
Ah yes since back trace feature is directly added to main.cpp file. Maybe we can move it to OMEditLIB.

But still if you add INCLUDEPATH to OMEditGUI instead of QMAKE_CXXFLAGS then this should work.

**Comment by AnHeuermann** [comment]
Nevermind, I got it working with only `INCLUDEPATH`. Sorry for the back and forth, I'm not used to the qt build stuff, so it was more guessing than knowing what to do.

**Review by adeas31 — APPROVED** [review]

**Pull request merged**